modules/virt-defining-a-watchdog.adoc

```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/advanced_vm_management/virt-configuring-a-watchdog.adoc
+
+[id="virt-defining-a-watchdog_{context}"]
+= Defining a watchdog
+
+.Procedure
+
+. Choose the watchdog action. This defines how the watchdog proceeds when the OS can no longer respond.
+
++
+.Available actions
+[horizontal]
+`poweroff`:: The virtual machine (VM) powers down, and the guest operating system (OS) cannot react.
+`reset`:: The VM reboots in place, and the guest OS cannot react. Because the length of time required for the guest OS to reboot can cause liveness probes to timeout, you should avoid their use.
+`shutdown`:: The VM powers off gracefully and becomes non-runnable.
+
+. Step 2? Create a YAML file and apply it to the cluster?
+. Step 3?
+
+The example below configures the i6300esb watchdog device on a Fedora VM with the poweroff action and exposes the device as `/dev/watchdog`.
+
+[source,yaml]
+----
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vmi-with-watchdog
+  name: vmi-with-watchdog
+spec:
+  domain:
+    devices:
+      watchdog:
+        name: mywatchdog
+        i6300esb:
+          action: "poweroff"
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 512M
+  terminationGracePeriodSeconds: 0
+  volumes:
+  - containerDisk:
+      image: kubevirt/fedora-cloud-container-disk-demo
+    name: containerdisk
+----
+
+This device can now be used by the watchdog binary. The following example will send a heartbeat every two seconds to `/dev/watchdog` and execute the defined action (in this case, `poweroff`) after four seconds without a heartbeat. This command must be run inside the guest.
```

Comment:
Maybe you can rewrite this slighlty. I think the important point is, that this action will be executed when the binary does not call `/dev/watchdog` anymore. It sounds a little like the binary would be responsible for the `poweroff` action.